napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+# Error handling
+
+N-API uses both return values and exceptions for error handling.  The sections
+which follow explain the approach for each case.
+
+## Return values
+
+All of the N-API functions share the same error handling pattern.  The
+return type of all API functions is ```napi_status```.
+
+The return value will be ```napi_ok``` if the request was successful and
+no uncaught exception was thrown. If an error occurred AND an exception
+was thrown, the napi_status value for the error will be returned.  If an
+exception was thrown, and no error occurred, ```napi_pending_exception```
+will be returned.
+
+In cases were a return value other than ```napi_ok``` or
```

Comment:
The triple backticks confuse the GitHub syntax highlighter. Single backticks should be used for inline code.